src/base/static/components/molecules/tag-editor.js

```diff
@@ -0,0 +1,155 @@
+  fontWeight: "normal",
+  transition: "all .5s ease",
+}));
+
+class TagEditor extends Component {
+  state = {
+    isFocused: false,
+    note: this.props.placeTag ? this.props.placeTag.note : "",
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  render() {
+    const isSelected = !!this.props.placeTag;
+
+    return (
+      <TagContainer
+        isSelected={isSelected}
+        backgroundColor={this.props.backgroundColor}
+        onClick={async () => {
+          // Toggle the placeTag.
+          if (isSelected) {
+            await mapseedApiClient.placeTags.delete(this.props.placeTag.url);
+            this.props.onDeletePlaceTag(this.props.placeTag.id);
+          } else {
+            const tagData = await mapseedApiClient.placeTags.create(
+              this.props.placeUrl,
+              {
+                note: this.state.note,
+                place: this.props.placeUrl,
+                tag: this.props.tag.url,
+              },
+            );
+            this.props.onCreatePlaceTag(tagData);
+          }
+        }}
+      >
+        <TagName displayName={this.props.displayName} isSelected={isSelected} />
+        {isSelected && (
+          <NoteBox
+            value={this.state.note}
+            placeholder={this.props.t("addNotePlaceholder")}
+            isFocused={this.state.isFocused}
+            onBlur={async () => {
+              // Save the note text on blur.
+              const tagData = await mapseedApiClient.placeTags.update(
+                this.props.placeTag.url,
+                {
+                  note: this.state.note,
+                },
+              );
+              this.props.onUpdateTagNote(tagData);
+              if (this._isMounted) {
+                // The blur event could fire after the component unmounts (for
+                // example if the user closes the content panel). Only update
+                // the local state if the component is still mounted.
+                this.setState({ isFocused: false });
+              }
+            }}
+            onFocus={() => {
+              this.setState({ isFocused: true });
+            }}
+            onClick={evt => {
+              evt.stopPropagation();
+            }}
+            onChange={evt => {
+              evt.preventDefault();
+              this.setState({
+                note: evt.target.value,
+              });
+            }}
+          />
+        )}
+      </TagContainer>
+    );
+  }
+}
+
+TagEditor.propTypes = {
+  backgroundColor: PropTypes.string,
+  placeTag: PropTypes.shape({
+    url: PropTypes.string.isRequired,
+    id: PropTypes.number.isRequired,
+    note: PropTypes.string,
+  }),
+  placeUrl: PropTypes.string.isRequired,
+  tag: PropTypes.shape({
+    id: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+    parent: PropTypes.number,
+    url: PropTypes.string.isRequired,
+  }).isRequired,
```

Comment:
I think it would be wise to make this a `tagPropType` variable, and export it from our Datasets duck. We should also include the `displayName` attribute.